zingo-rpc/src/blockcache/utils.rs

```diff
@@ -0,0 +1,143 @@
+//! Blockcache utility functionality.
+
+use byteorder::{LittleEndian, ReadBytesExt};
+use std::io::{Cursor, Read};
+
+use crate::jsonrpc::connector::JsonRpcConnectorError;
+
+/// Parser Error Type.
+#[derive(Debug)]
+pub enum ParseError {
+    /// Io Error
+    Io(std::io::Error),
+    /// Invalid Data Error.
+    InvalidData(String),
+}
+
+impl From<std::io::Error> for ParseError {
+    fn from(err: std::io::Error) -> ParseError {
+        ParseError::Io(err)
+    }
+}
+
+impl std::fmt::Display for ParseError {
+    fn fmt(&self, f: &mut std::fmt::Formatter<'_>) -> std::fmt::Result {
+        match self {
+            ParseError::Io(err) => write!(f, "IO Error: {}", err),
+            ParseError::InvalidData(msg) => write!(f, "Invalid Data Error: {}", msg),
+        }
+    }
+}
```

Comment:
I would enthusiastically recommend the crate [thiserror](https://docs.rs/thiserror/latest/thiserror/).
I don't know if there's an easy way to do the from JsonConnector error as a string like that, but the entire chunk I have selected could become
```suggestion
/// Parser Error Type.
#[derive(Debug, thiserror::Error)]
pub enum ParseError {
    #[error("IO Error: {0}")]
    Io(#[from] std::io::Error),
    #[error("Invalid Data Error: {0}")]
    InvalidData(String),
}
```

Reply:
We're currently in the process of switching to thiserror in zingolib. It's not on all our error types, but we've a bunch of it going on for example [here](https://github.com/zingolabs/zingolib/blob/dev/zingolib/src/lightclient/send.rs#L166)

Reply:
Ah yes oscar showed me this just after i coded these errors. ive left them in for now while im chasing bugs but will refactor ParseError and JsonRpcConnectorError using ThisError before marking as ready for review.

Reply:
Ive converted all inetrnal error types to use thiserror in [4e91fff].